### PR TITLE
Closes #434 #463 - Wait for all notifications to be processed.  Stop exc...

### LINKED ISSUE #434: Wait for all notifications to be processed
While trying to completely automate running the stress test on the cluster, I realized I really need a fluo command line command to wait for all notifications to be processed.   Scanning the notification locality groups is not good enough because of a race condition.  The race conditions w/ simply scanning is that notifications not yet seen by the scan can cause notifications to be written behind the scan.   

To check for this race condition I am thinking a wait command could do something like the following :

```java
  while(true){
    int ts1 = oracle.allocateTimeStamp();
    int ntfyCount = countNotifications();
    int ts2 = oracle.allocateTimeStamp();
    if(ntfyCount == 0 && ts1 == (ts2 -1)){
      break;
    }
    int sleepTime = f(ntfyCount);
    sleep(sleepTime);
  }
```

### PR DESCRIPTION
**Pull request by mikewalch**
...eptions from being logged when OracleClient is closed

### CODE REVIEW

**Comment by keith-turner** [comment]
what motivated all of the pom changes?

File: modules/cluster/src/main/java/io/fluo/cluster/runner/AppRunner.java [hunk 1]
```diff
@@ -254,4 +222,55 @@ public static void classpath(String scriptName, String fluoHomeDir, String[] arg
 
     System.out.println(classpath.toString());
   }
+
+  private long calculateSleep(long notifyCount, long numWorkers) {
+    long sleep = notifyCount / numWorkers / 100;
+    if (sleep < MIN_SLEEP_SEC) {
+      return MIN_SLEEP_SEC;
+    }
+    return sleep;
```
Comment (keith-turner) [inline modules/cluster/src/main/java/io/fluo/cluster/runner/AppRunner.java]: May want to have a max also... maybe 5 or 10 mins.

File: modules/cluster/src/test/java/io/fluo/cluster/runner/MiniAppRunnerIT.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+import io.fluo.core.TestTransaction;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class MiniAppRunnerIT extends ITBaseImpl {
+
+  @Test
+  public void testSimple() {
+    TestTransaction tx1 = new TestTransaction(env);
+    tx1.mutate().row("row1").fam("fam1").qual("q1").set("v1");
```
Comment (keith-turner) [inline modules/cluster/src/test/java/io/fluo/cluster/runner/MiniAppRunnerIT.java]: will this create notifications?

**Comment by mikewalch** [comment]
One reason for the pom changes was that I needed to to start up mini-accumulo in the fluo-cluster module for the new  MiniAppRunnerIT.  I also tried to use only logback in the fluo-cluster module.  However, mini-accumulo requires log4j.  Therefore, you really can't exclude one or the other.  Therefore, I removed the log4j exclusions from the pom.  These exclusion are not necessary any more as the maven assembly plugin (see bin.xml) separates log4j & logback jars into different directories.

File: modules/cluster/src/test/java/io/fluo/cluster/runner/MiniAppRunnerIT.java (hunk 2 above)
Comment (mikewalch) [inline modules/cluster/src/test/java/io/fluo/cluster/runner/MiniAppRunnerIT.java]: It doesn't create notificaitons.   Do you think I should create them?  They will probably be processed by the time `runner.waitUntilFinished()` is called.

File: modules/cluster/src/main/java/io/fluo/cluster/runner/AppRunner.java (hunk 1 above)
Comment (mikewalch) [inline modules/cluster/src/main/java/io/fluo/cluster/runner/AppRunner.java]: Ok will do

File: modules/cluster/src/test/java/io/fluo/cluster/runner/MiniAppRunnerIT.java (hunk 2 above)
Comment (mikewalch) [inline modules/cluster/src/test/java/io/fluo/cluster/runner/MiniAppRunnerIT.java]: I will create some. It can't hurt.

Comment (keith-turner) [inline modules/cluster/src/test/java/io/fluo/cluster/runner/MiniAppRunnerIT.java]: > They will probably be processed by the time runner.waitUntilFinished() is called.

Maybe have the observer that processes them sleep??

File: modules/core/src/main/java/io/fluo/core/oracle/OracleClient.java [hunk 3]
```diff
@@ -105,9 +106,10 @@ public void run() {
           connect();  
         }
         doWork();
-
       } catch (Exception e) {
-        e.printStackTrace();
+        if (!closed.get()) {
+          log.error("Exception occurred in run() method", e);
+        }
```
Comment (keith-turner) [inline modules/core/src/main/java/io/fluo/core/oracle/OracleClient.java]: Could log at debug if closed.  Hopefully that would not normally show up.

File: modules/core/src/main/java/io/fluo/core/oracle/OracleClient.java [hunk 4]
```diff
@@ -190,12 +192,11 @@ private void doWork() {
             tr.cdl.countDown();
           }
         } catch (InterruptedException e) {
-          if(closed){
-            return;
+          if(!closed.get()){
+            log.error("InterruptedException occurred in doWork() method", e);
```
Comment (keith-turner) [inline modules/core/src/main/java/io/fluo/core/oracle/OracleClient.java]: could also log debug if closed here

Comment (mikewalch) [inline modules/core/src/main/java/io/fluo/core/oracle/OracleClient.java]: fixed in 4a28775685

File: modules/core/src/main/java/io/fluo/core/oracle/OracleClient.java (hunk 3 above)
Comment (mikewalch) [inline modules/core/src/main/java/io/fluo/core/oracle/OracleClient.java]: fixed in 4a28775685

File: modules/cluster/src/test/java/io/fluo/cluster/runner/MiniAppRunnerIT.java (hunk 2 above)
Comment (mikewalch) [inline modules/cluster/src/test/java/io/fluo/cluster/runner/MiniAppRunnerIT.java]: fixed 4a28775685

File: modules/cluster/src/main/java/io/fluo/cluster/runner/AppRunner.java (hunk 1 above)
Comment (mikewalch) [inline modules/cluster/src/main/java/io/fluo/cluster/runner/AppRunner.java]: fixed in 4a28775685

**Comment by keith-turner** [comment]
+1

**Pull request merged**